src/nacl/public.py

```diff
@@ -72,27 +72,46 @@ class PrivateKey(encoding.Encodable, StringFixer, object):
     :param encoder: The encoder class used to decode the given keys
 
     :cvar SIZE: The size that the private key is required to be
+    :cvar SEED_SIZE: The size that the seed used to generate the
+                     private key is required to be
     """
 
     SIZE = nacl.bindings.crypto_box_SECRETKEYBYTES
+    SEED_SIZE = nacl.bindings.crypto_box_SEEDBYTES
 
     def __init__(self, private_key, encoder=encoding.RawEncoder):
         # Decode the secret_key
         private_key = encoder.decode(private_key)
-        if not isinstance(private_key, bytes):
-            raise exc.TypeError(
-                "PrivateKey must be created from a 32 byte seed")
-
-        # Verify that our seed is the proper size
-        if len(private_key) != self.SIZE:
-            raise exc.ValueError(
-                "The secret key must be exactly %d bytes long" % self.SIZE)
+        # verify the given secret key type and size are correct
+        if not (isinstance(private_key, bytes)
+                and len(private_key) == self.SIZE):
+            raise exc.TypeError(("PrivateKey must be created from a {0} "
+                                 "bytes long raw secret key").format(self.SIZE)
+                                )
 
         raw_public_key = nacl.bindings.crypto_scalarmult_base(private_key)
 
         self._private_key = private_key
         self.public_key = PublicKey(raw_public_key)
 
+    @classmethod
+    def from_seed(cls, seed, encoder=encoding.RawEncoder):
+        # decode the seed
+        seed = encoder.decode(seed)
+        # Verify the given seed type and size are correct
+        if not (isinstance(seed, bytes) and len(seed) == cls.SEED_SIZE):
+            raise exc.TypeError(("PrivateKey seed must be a {0} bytes long "
+                                 "binary sequence").format(
+                                                           cls.SEED_SIZE)
+                                )
+        # generate a raw keypair from the given seed
+        raw_pk, raw_sk = nacl.bindings.crypto_box_seed_keypair(seed)
+        # create a PrivateKey instance from the raw keypair
+        prvk = super(cls.__class__, cls).__new__(cls)
```

Comment:
This is just from a dislike of repeated work... scalarmult is fast, but why do it twice in a row?
If you think readabiliy counts more than that, I'll go back to using the standard constructor.